src/ApiCompat/Microsoft.DotNet.ApiCompatibility/ApiComparer.cs

```diff
@@ -47,53 +46,42 @@ public class ApiComparer : IApiComparer
             string? leftName = null,
             string? rightName = null)
         {
-            if (left == null)
-            {
-                throw new ArgumentNullException(nameof(left));
-            }
-
-            if (right == null)
-            {
-                throw new ArgumentNullException(nameof(right));
-            }
-
             AssemblyMapper mapper = new(GetComparingSettingsCore(leftName, rightName != null ? new[] { rightName } : null));
             mapper.AddElement(left, ElementSide.Left);
             mapper.AddElement(right, ElementSide.Right);
 
             DifferenceVisitor visitor = new();
             visitor.Visit(mapper);
-            return visitor.DiagnosticCollections.First();
+            return visitor.DiagnosticCollections[0];
         }
 
         /// <inheritdoc />
         public IEnumerable<(MetadataInformation left, MetadataInformation right, IEnumerable<CompatDifference> differences)> GetDifferences(ElementContainer<IAssemblySymbol> left,
             IList<ElementContainer<IAssemblySymbol>> right)
         {
             int rightCount = right.Count;
-            AssemblyMapper mapper = new(new ComparingSettings(), rightSetSize: rightCount);
-            mapper.AddElement(left.Element, ElementSide.Left);
 
+            // Retrieve the right names
             string[] rightNames = new string[rightCount];
             for (int i = 0; i < rightCount; i++)
             {
-                ElementContainer<IAssemblySymbol> element = right[i];
-                rightNames[i] = element.MetadataInformation.DisplayString;
-                mapper.AddElement(element.Element, ElementSide.Right, i);
+                rightNames[i] = right[i].MetadataInformation.DisplayString;
```

Comment:
Constructing the rightNames before initializing the AssemblyMapper to avoid instantiating a throw-away ComparingSettings object.